mindee/product/invoice_splitter/invoice_splitter_v1_document.py

```diff
@@ -19,17 +19,17 @@ def __init__(self, raw_prediction: StringDict) -> None:
 
         :param raw_prediction: Raw prediction from HTTP response
         """
+        invoice_page_groups = []
         if (
             "invoice_page_groups" in raw_prediction
             and len(raw_prediction["invoice_page_groups"]) > 0
         ):
-            self.invoice_page_groups = [
-                InvoiceSplitterV1PageGroup(prediction)
-                for prediction in raw_prediction["invoice_page_groups"]
-            ]
+            for prediction in raw_prediction["invoice_page_groups"]:
+                invoice_page_groups.append(InvoiceSplitterV1PageGroup(prediction))
+        self.invoice_page_groups = invoice_page_groups
 
     def __str__(self) -> str:
         page_group_str = ":Invoice Page Groups:"
         for page_group in self.invoice_page_groups:
             page_group_str += f"\n  {str(page_group)}"
-        return clean_out_string(page_group_str)
+        return clean_out_string(page_group_str).rstrip()
```

Comment:
Not really, it just removes spaces before carriage returns, which isn't all that useful in hindsight. I think I might add the `rstrip()` to it directly so that at least the name makes sense.